### PR TITLE
added cell verification

### PR DESCRIPTION
**Pull request by Tombez**

### CODE REVIEW

File: Cells.js [hunk 1]
```diff
@@ -90,28 +90,33 @@ export const data = {
...
+})(data);
+
 export function canPhase(self, other) {
   if(self == null || other == null) return false;
   const selfData = data[self.id];
   const otherData = data[other.id];
   return selfData.hasGravity && otherData.hasGravity && selfData.density > otherData.density;
 }
 
-export function convert(self, other) {
```
Comment (kamoroso94) [inline Cells.js:100]: Why did you delete the function?

File: Cells.js [hunk 2]
```diff
@@ -90,28 +90,33 @@ export const data = {
   }
 };
 
+(function verifyCells(cells) {
```
Comment (kamoroso94) [inline Cells.js]: This seems more like a test

File: tick.js [hunk 3]
```diff
@@ -27,14 +27,19 @@ export default function tick(event) {
   // spreading
   grid.forEach((cell1, [x1, y1]) => {
     if(visited.has(cell1)) return;
-    if(!Cells.data[cell1.id].hasOwnProperty('conversions')) return;
+    const cell1Data = Cells.data[cell1.id];
```
Comment (kamoroso94) [inline tick.js:30]: I like how you broke up this long line.

File: Cells.js (hunk 1 above)
Comment (Tombez) [reply Cells.js:100]: It's not needed anymore.

File: Cells.js (hunk 2 above)
Comment (Tombez) [reply Cells.js]: What are you suggesting I change?

File: Cells.js (hunk 1 above)
Comment (kamoroso94) [reply Cells.js:100]: I guess the question is why did you relocate the functionality?

File: Cells.js (hunk 2 above)
Comment (kamoroso94) [reply Cells.js]: Maybe it can be in a test script instead.

File: Cells.js (hunk 1 above)
Comment (Tombez) [reply Cells.js:100]: Because now it runs once on startup, and it's guaranteed to run for every conversion, so there isn't the possibility of typoing a conversion, and then not testing it.

File: Cells.js (hunk 2 above)
Comment (Tombez) [reply Cells.js]: Well it also fills in missing 'self' properties in the conversions, so it wouldn't really work as a unit test.

Comment (kamoroso94) [reply Cells.js]: That seems a bit extra, including the self change when it is kind of redundant, no?

File: Cells.js (hunk 1 above)
Comment (kamoroso94) [reply Cells.js:100]: This goes towards the other section as well: I don't think we need the test code in the production code.

File: tick.js [hunk 4]
```diff
@@ -27,14 +27,19 @@ export default function tick(event) {
...
+    if(!cell1Data.hasOwnProperty('conversions')) return;
 
     for(const [cell2, [x2, y2]] of grid.neighborEntries(x1, y1)) {
       // not diagonal neighbors
       if(Math.abs(x1 - x2) + Math.abs(y1 - y2) != 1) continue;
       if(Math.random() >= Cells.SPREAD_RATE) continue;
 
-      Cells.convert(cell1, cell2);
+      const conversion = cell1Data.conversions[cell2.id];
```
Comment (kamoroso94) [inline tick.js:38]: I guess this is much simpler.

File: Cells.js (hunk 2 above)
Comment (Tombez) [reply Cells.js]: It's faster than checking it on every conversion.

Comment (kamoroso94) [reply Cells.js]: I think you should remove this and possibly reintroduce it into a testing script.

File: tick.js (hunk 4 above)
Comment (kamoroso94) [reply tick.js:38]: I think you should have an if check to see if `self` and possibly `other` are defined before assigning those values.  Otherwise we can hardcode it into the JSON if you'd rather avoid the check.

**Comment by kamoroso94** [comment]
If you can make the suggested changes I can merge this.

File: tick.js (hunk 4 above)
Comment (Tombez) [reply tick.js:38]: I hardcoded it into the JSON.

File: Cells.js (hunk 2 above)
Comment (Tombez) [reply Cells.js]: I moved it to a testing script.

File: index.html [hunk 5]
```diff
@@ -33,6 +33,14 @@
     />
 
     <!-- Custom -->
+    <link
+      rel="prefetch"
+      href="materials.json"
+      as="fetch"
+      type="text/json"
+      crossorigin="anonymous"
+      onload="fetch('materials.json').then(res=>res.json()).then(obj=>window.materials=obj)"
```
Comment (kamoroso94) [inline index.html:42]: This is too hacky and exposes the cell data in the global scope, revert back to regular object for now.

File: conversions-test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+const fs = require('fs');
```
Comment (kamoroso94) [inline conversions-test.js:1]: I don't think we're ready to write tests yet, but if so, move it to a test directory.

File: Cells.js [hunk 7]
```diff
@@ -3,92 +3,7 @@
 export const SIZE = 4;
 export const SPREAD_RATE = 1/16;
 
-export const data = {
```
Comment (kamoroso94) [inline Cells.js:6]: Revert back to regular object for now.

**Review by kamoroso94 — CHANGES_REQUESTED** [review]
Exposing the cell data in the global scope is not acceptable.

File: index.html (hunk 5 above)
Comment (Tombez) [reply index.html:42]: What if I delete it right after Cell.js gets a reference to it?
